src/pytorch_lightning/loggers/wandb.py

```diff
@@ -450,13 +451,13 @@ def save_dir(self) -> Optional[str]:
 
     @property
     def name(self) -> Optional[str]:
```

Comment:
I think it's a little off now since `name` argument and property are different. Instead, should we add/rename `project_name`?

Reply:
I can't fix it in another way. The name here is tied to what is being required by the checkpointing logic, as explained in #14188. 

The only other way would be to hack an "isinstance check" in the ModelCheckpoint class and change the behavior for just wandb, but considering that #14188 is looking promising, I want to avoid this at all cost. 

Besides, it is also the same for the default logger, TensorBorad, where name does not represent the individual run's name, but the name of the entire "project'.

Reply:
#14188 looks good, but considering the logger individually, this is a little inconsistent since we have both `name` and `project_name` as arguments. I think we should have both these properties,  or which ever is required after #14188.

> Besides, it is also the same for the default logger, TensorBorad, where name does not represent the individual run's name, but the name of the entire "project'.

a wandb user might not care about other loggers at all.

Reply:
There will always be terminology in the 3rd party apis that does not mean the same thing in PL. The name property is on the base interface and I can't change it in just one logger. The discussion needs to be offloaded, I cannot address this here. If you want to add a project name property to the WandbLogger, feel free to do that. But at the moment, the implementation of "name" can't change, not until we have https://github.com/Lightning-AI/lightning/issues/14188

Reply:
will create an issue to discuss this.